File: backend/app/domain/repositories/vaccination_repository.py
from abc import ABC, abstractmethod
from datetime import date
from uuid import UUID

from app.domain.entities.vaccination import Vaccination


class VaccinationRepository(ABC):
    @abstractmethod
    async def get_by_baby_id(self, baby_id: UUID) -> list[Vaccination]:
        ...

    @abstractmethod
    async def get_upcoming(self, baby_id: UUID, within_days: int = 30) -> list[Vaccination]:
        ...

    @abstractmethod
    async def save(self, v: Vaccination) -> Vaccination:
        ...

    @abstractmethod
    async def mark_administered(
        self,
        id: UUID,
        administered_date: date,
        hospital_name: str | None,
    ) -> Vaccination:
        ...

    @abstractmethod
    async def delete(self, id: UUID) -> None:
        ...

File: zzippu_ios/zzippu/Domain/Repositories/VaccinationRepository.swift
// Domain/Repositories/VaccinationRepository.swift
// Foundation only — SwiftUI/SwiftData import 금지.

import Foundation

protocol VaccinationRepository {
    func list(babyId: UUID) async throws -> [Vaccination]
    func markAdministered(
        babyId: UUID,
        id: UUID,
        administeredDate: Date,
        hospitalName: String?
    ) async throws -> Vaccination
}
